Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -1010,9 +1010,12 @@ void RenderObject::issueRepaint(std::optional<LayoutRect> partialRepaintRect, Cl
         return;
 
     LayoutRect repaintRect;
-    if (partialRepaintRect)
-        repaintRect = clipRepaintToContainer == ClipRepaintToContainer::No ? *partialRepaintRect : computeRectForRepaint(*partialRepaintRect, repaintContainer.renderer);
```

Comment:
@smfr thanks for pushing on this, much appreciated. The root cause of the bug was simply using `*partialRepaintRect` which is wrong.